pkg/bundle/sync.go

```diff
@@ -777,3 +779,38 @@ func (b *bundle) migrateConfigMapToApply(ctx context.Context, obj client.Object,
 	cm.SetManagedFields(managedFields)
 	return true, b.directClient.Update(ctx, &cm)
 }
+
+// remove duplicate certificates from bundles
+func deduplicateBundles(bundles []string) []string {
+	var block *pem.Block
+
+	var certificatesHashes = make(map[[32]byte]struct{})
+	var dedupCerts []string
+
+	for _, cert := range bundles {
+		certBytes := []byte(cert)
+	LOOP:
+		for {
+			block, certBytes = pem.Decode([]byte(certBytes))
```

Comment:
I have a feeling we are decoding/encoding certs in bundles multiple times already, and this adds another round of decode/encode. What about using a "Cert Pool" as an intermediate format between functions - instead of `[]string` or `[]byte`? We could fix this in a follow-up PR. Just wanted to note it here.

Reply:
Yes, you are right, however there is no way to parse bundle of cert which was converted into string, it come out from [here](https://github.com/cert-manager/trust-manager/blob/cd860d6da4580e9aa1d193b33d726e4a01ad3af0/pkg/bundle/sync.go#L124)

The issue is that some items might consist  more than one cert
```
-----BEGIN CERTIFICATE----
some data 1
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE----
some data 1
-----END CERTIFICATE-----
-----BEGIN CERTIFICATE----
some data 2
-----END CERTIFICATE-----
```